hooks/useTokenBalance.tsx

```diff
@@ -0,0 +1,47 @@
+import { TokenListItem } from "@/hooks/useTokenList";
+import { BigNumber } from "@ethersproject/bignumber";
+import { multicall } from "@wagmi/core";
+import useSWR from "swr";
+import { useAccount, useNetwork } from "wagmi";
+
+export const ERC20_ABI = [
+  "function balanceOf(address owner) view returns (uint256 balance)",
+];
+
+const useTokenList = ({ tokenList }: { tokenList: TokenListItem[] }) => {
+  const { chain } = useNetwork();
+  const { address } = useAccount();
+
+  const chainId = chain?.id || 1; // default to mainnet if no chain id
+  const { isLoading, error, data } = useSWR(
+    `userTokenList-${chainId}-${tokenList.length}`,
+    async () => {
```

Comment:
Is my understand of `swr` correct that by default this re-fetches the data every 3 seconds? That seems reasonable to me

Reply:
I think the default is 2 seconds, but we can add a custom configuration if that is too often. 

Default configuration: https://github.com/vercel/swr/blob/main/_internal/utils/config.ts#L48

Reply:
That seems fine for now, can always change it later